File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;

/**
 *
 * @author pc
 */
public class CasillaCalle extends Casilla {
    
    private TituloPropiedad tituloPropiedad;
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        tituloPropiedad=titulo;
    }
    
    @Override
     public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            Jugador jugador=todos.get(actual);
            
            if(!tituloPropiedad.tienePropietario())
                jugador.puedeComprarCasilla();
            else{
                tituloPropiedad.tramitarAlquiler(jugador);
                }
        }
    }
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    @Override
    public String toString() {
        return "Casilla: "+ this.getNombre() +"\n      TituloPropiedad: " + tituloPropiedad.toString();
    }
     
}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class CasillaCalle < Casilla
  
  
  def initialize(nombre,titulo)
    super(nombre)
    @tituloPropiedad=titulo
  end
  def recibeJugador(actual,todos)
    if(jugadorCorrecto(actual, todos))
      super(actual,todos)
      jugador=todos.at(actual)
      
      if(@tituloPropiedad.tienePropietario()==false)
        jugador.puedeComprarCasilla
      else
        @tituloPropiedad.tramitarAlquiler(jugador)
      end
    end
  end
  public
  def getTituloPropiedad()
    return @tituloPropiedad 
  end
  
  public
  def toString
    msg="\n\nCasilla: #{@nombre}"  + @tituloPropiedad.toString +  "\n\n"
    return msg
  end
  
end
end